src/main/java/org/apache/commons/imaging/formats/tiff/photometricinterpreters/fp/PaletteEntryForRange.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.imaging.formats.tiff.photometricinterpreters.fp;
+
+import java.awt.Color;
+
+/**
+ * Provides a palette entry for colors associated with a range of values. The
+ * return value will be interpolated between the minimum and maximum value for
+ * this entry.
+ * <p>
+ * In keeping with the conventions of many Geographic Information Systems (GIS)
+ * and art applications, this instance "covered" values in the range v0 &le; f
+ * &lt; v1. Thus, a value that exactly matches the upper bound of the range is
+ * not considered "covered".
+ */
+public class PaletteEntryForRange implements IPaletteEntry {
+
+    private final float v0;
+    private final float v1;
+    private final float deltaV;
+    private final float r0;
+    private final float r1;
+    private final float g0;
+    private final float g1;
+    private final float b0;
+    private final float b1;
+    private final float a0;
+    private final float a1;
+    private final float deltaA;
+    private final float deltaR;
+    private final float deltaG;
+    private final float deltaB;
```

Comment:
Probably nit-picking. I think the variables above are related to variable1, variable0, red1, red2, green1, green2, etc. But could be wrong.

In case these variables appear somewhere like debugger/profiler/log/stacktrace/etc, it would be better to have a better name I think. Is it something that could be changed @gwlucastrig ? (I acknowledge sometimes the best name is the simplest like these, so just asking if you have any suggestion that could make it easier for other contributors)

Maybe as we are talking about range, the v0 and v1 could be left, right. Or from, to. Or fromInclusive, toInclusive, etc. Something that is easy for another developer used to other "ranges" to associate with?

Reply:
We could just get rid of the "delta" variables and change the places where it's used such as
     int r = (int) (t * deltaR + r0 + 0.5);
To 
     int r = (int) (t * (r1-r0) + r0 + 0.5);

I don't think the first form actually saves any processing time.  It's just the way I happened to code it.

Reply:
>I don't think the first form actually saves any processing time. It's just the way I happened to code it.

All good leaving delta variables. What about the names of variables? Do you have an opinion on this?